### PR TITLE
Update main to include merged interop changes

### LINKED ISSUE #7191: [Merged Interops] Microsoft.DotNet.IBCMerge
We had to change the way we did `/ibc` since now we're fetching `Microsoft.DotNet.IBCMerge` from a new feed which also requires Nuget.config file.

The current solution is perhaps a better method of doing this and given our lack of optprof we weren't really using ibc so it's hard to verify. I'll need to do a deeper analysis on this and clean up the code surrounding this

### PR DESCRIPTION
**Pull request by dmonroym**
This merges all the work associated with merged interops and brings up to parity what is in 17.0 Preview 1. This also brings the package version up to `17.0.1` to be able to differentiate between P1 and P2 packages.

I decided to track the issues this opens separately and fix those afterwards. It opens:
 - #7191: The `/ibc` option is disabled and set to `/no-ibc`. Currently this doesn't affect our build since we don't use optprof but I had to introduce some changes surrounding the build of `RestoreIBCMerge.csproj` including temporarily adding a Nuget.Config file. This file is solely used for this project as we need to retrieve the package from a private feed so we can't include it in packages.targets
 - #7192 Creates some functional problems and is being addressed. Our tests caught this and they're currently disabled. This is fixed in #7234 and will forward integrate after this is merged as it was made into preview 1.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/project-system/pull/7233)

### CODE REVIEW

**Comment by dmonroym** [comment]
Validated this with https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=4771137&view=results

File: build/import/Versions.props [hunk 1]
```diff
@@ -3,7 +3,7 @@
   <PropertyGroup>
     <!-- IMPORTANT: These assembly versions must always be 3 digits -->
     <MinimumRequiredVSVersion>16.8.1</MinimumRequiredVSVersion> <!-- We should try to keep this up to date with the highest version available on build machines -->
-    <ProjectSystemVersion>16.10.0</ProjectSystemVersion> <!-- This should always match with the VS release we're targeting -->
+    <ProjectSystemVersion>17.0.1</ProjectSystemVersion> <!-- This should always match with the VS release we're targeting -->
```
Comment (tmeschter) [inline build/import/Versions.props:6]: Should we bump this to 17.0.2 for Preview 2 and beyond?

File: src/Microsoft.VisualStudio.Editors/Microsoft.VisualStudio.Editors.vbproj [hunk 2]
```diff
@@ -281,9 +281,7 @@
       <DependentUpon>PackagePropPage.vb</DependentUpon>
       <SubType>UserControl</SubType>
     </Compile>
-    <Compile Update="PropPages\PackagePropPage.vb">
-      <SubType>UserControl</SubType>
```
Comment (tmeschter) [inline src/Microsoft.VisualStudio.Editors/Microsoft.VisualStudio.Editors.vbproj:285]: Do we know why we lost the `SubType`s?

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/LaunchProfilesDebugLaunchProvider.cs [hunk 3]
```diff
@@ -262,9 +262,9 @@ internal static VsDebugTargetInfo4 GetDebuggerStruct4(IDebugLaunchSettings info)
             {
                 var processStartupInfo = new VsDebugStartupInfo
                 {
-                    hStdInput = unchecked((uint)info.StandardInputHandle.ToInt32()),
-                    hStdOutput = unchecked((uint)info.StandardOutputHandle.ToInt32()),
-                    hStdError = unchecked((uint)info.StandardErrorHandle.ToInt32()),
+                    hStdInput = unchecked((IntPtr)(uint)info.StandardInputHandle.ToInt32()),
```
Comment (tmeschter) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/LaunchProfilesDebugLaunchProvider.cs:265]: Can we go straight from `info.StandardInputHandle` to `IntPtr` without going though Int32 and uint first?

That is, are we able to simply these expressions somehow?

**Review by tmeschter — APPROVED** [review]
Approved with comments.

File: build/import/Versions.props (hunk 1 above)
Comment (dmonroym) [reply build/import/Versions.props:6]: Preview 1 is 17.0.0 so I just bumped it up one but would it make more sense to go to 17.0.2?

File: src/Microsoft.VisualStudio.Editors/Microsoft.VisualStudio.Editors.vbproj (hunk 2 above)
Comment (dmonroym) [reply src/Microsoft.VisualStudio.Editors/Microsoft.VisualStudio.Editors.vbproj:285]: Shouldn't have, thanks for catching that. I'll fix in next iteration.

File: build/import/Versions.props (hunk 1 above)
Comment (tmeschter) [reply build/import/Versions.props:6]: No, that's fine. Just ensuring that they are unique (and incrementing) is enough. I don't want to have more version numbers than necessary. :-)

File: src/Microsoft.VisualStudio.AppDesigner/PropPages/PropPageUserControlBase.vb [hunk 4]
```diff
@@ -2833,7 +2835,9 @@ NextControl:
 
             'First see if it is already in the project
             For Each ProjectItem As EnvDTE.ProjectItem In ProjectItems
-                If ProjectItem.FileNames(1).Equals(FileName, StringComparison.OrdinalIgnoreCase) Then
+#Disable Warning BC42025 ' Access of shared member, constant member, enum member or nested type through an instance
+                If ProjectItem.get_FileNames(1).Equals(FileName, StringComparison.OrdinalIgnoreCase) Then
+#Enable Warning BC42025 ' Access of shared member, constant member, enum member or nested type through an instance
```
Comment (dmonroym) [inline src/Microsoft.VisualStudio.AppDesigner/PropPages/PropPageUserControlBase.vb:2840]: These warning wrappers should be removed after #7235 is integrated. Tracked by #7257

File: src/Microsoft.VisualStudio.Editors/MyApplication/MyApplicationProperties.vb [hunk 5]
```diff
@@ -669,7 +671,9 @@ Namespace Microsoft.VisualStudio.Editors.MyApplication
             Get
                 'First see if it is already in the project
                 For Each ProjectItem As ProjectItem In ProjectDesignerProjectItem.ProjectItems
-                    If ProjectItem.FileNames(1).Equals(MyAppFileNameWithPath, StringComparison.OrdinalIgnoreCase) Then
+#Disable Warning BC42025 ' Access of shared member, constant member, enum member or nested type through an instance
```
Comment (dmonroym) [inline src/Microsoft.VisualStudio.Editors/MyApplication/MyApplicationProperties.vb:674]: Also tracked by #7257

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/LaunchProfilesDebugLaunchProvider.cs (hunk 3 above)
Comment (dmonroym) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/LaunchProfilesDebugLaunchProvider.cs:265]: Yes, in fact `info.StandardInputHandle` is an `IntPtr` so none of these are needed.

**Pull request merged**